tools/sigclient/pkg/ingest/k8s_generator.go

```
@@ -77,163 +95,180 @@ func (g *K8sGenerator) GetLogLine() ([]byte, error) {
 func (g *K8sGenerator) GetRawLog() (map[string]interface{}, error) {
 	g.metricCounter++
 
-	if g.bothMetrics {
+	// Generate base timestamp
+	baseTs := time.Now().Unix()
+
+	var metrics []map[string]interface{}
+
+	switch g.metricType {
+	case GenerateBothMetrics:
 		// Generate both KSM and node-exporter metrics
-		ksmMetric := g.generateKSMMetric()
-		nodeMetric := g.generateNodeExporterMetric()
-		return map[string]interface{}{
-			"ksm_metric":  ksmMetric,
-			"node_metric": nodeMetric,
-		}, nil
-	}
+		metrics = append(metrics, g.generateKSMMetric(baseTs))
+		metrics = append(metrics, g.generateNodeExporterMetric(baseTs))
+	case GenerateKSMOnly:
+		// Generate KSM only
+		metrics = append(metrics, g.generateKSMMetric(baseTs))
 
-	if g.ksmOnly {
-		return g.generateKSMMetric(), nil
-	}
+	case GenerateNodeExporterOnly:
+		// Generate node-exporter only
+		metrics = append(metrics, g.generateNodeExporterMetric(baseTs))
+
+	default:
+		// Default alternating behavior
+		if g.metricCounter%2 == 0 {
+			metrics = append(metrics, g.generateKSMMetric(baseTs))
+		} else {
+			metrics = append(metrics, g.generateNodeExporterMetric(baseTs))
+		}
```

I don't think there's a reason to do this. You can just return an error, unless I'm missing something